network-libp2p/src/network.rs

```diff
@@ -1737,6 +1758,46 @@ impl NetworkInterface for Network {
         }
     }
 
+    async fn get_peers_by_services(
+        &self,
+        services: Services,
+    ) -> Result<Vec<Self::PeerId>, NetworkError> {
+        let (output_tx, output_rx) = oneshot::channel();
+        let connected_peers = self.get_peers();
+        let mut filtered_peers = vec![];
+
+        // First we try to get the connected peers that support the desired services
+        for peer_id in connected_peers.iter() {
+            if let Some(peer_info) = self.get_peer_info(*peer_id) {
+                if peer_info.get_services().contains(services) {
+                    filtered_peers.push(*peer_id);
+                }
+            }
+        }
+
+        // If we dont have any connected peer that support the desired services,
```

Comment:
```suggestion
        // If we don't have any connected peer that support the desired services,
```

Reply:
And I thought we wanted to return an `optimal_num_peers` that the network always try to return if there are such number of peers that provides the required services